src/v/security/authorizer.h

```diff
@@ -41,11 +42,18 @@ class authorizer final {
     // allow operation when no ACL match is found
     using allow_empty_matches = ss::bool_class<struct allow_empty_matches_type>;
 
-    authorizer()
-      : authorizer(allow_empty_matches::no) {}
-
-    explicit authorizer(allow_empty_matches allow)
-      : _allow_empty_matches(allow) {}
+    authorizer(
+      std::function<config::binding<std::vector<ss::sstring>>()> superusers_cb)
```

Comment:
nit: lambda will still auto-convert into std::function so explicit on single-arg ctor here is still ok.